backup/src/main/java/io/camunda/zeebe/backup/management/BackupServiceImpl.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.backup.management;
+
+import io.camunda.zeebe.backup.api.BackupStore;
+import io.camunda.zeebe.scheduler.ConcurrencyControl;
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+final class BackupServiceImpl {
+  private final int nodeId;
+  private final int partitionId;
+
+  private final int numberOfPartitions;
+
+  private final Set<InProgressBackup> backupsInProgress = new HashSet<>();
+  private final BackupStore backupStore;
+
+  BackupServiceImpl(
+      final int nodeId,
+      final int partitionId,
+      final int numberOfPartitions,
+      final BackupStore backupStore) {
+    this.nodeId = nodeId;
+    this.partitionId = partitionId;
+    this.numberOfPartitions = numberOfPartitions;
+    this.backupStore = backupStore;
+  }
+
+  void close() {
+    backupsInProgress.forEach(InProgressBackup::close);
+  }
+
+  ActorFuture<Void> takeBackup(
```

Comment:
:question: Do we want some sort of timeout for taking backups? So that stuck backups don't accumulate in `backupsInProgress`

Reply:
Good point. Let me think about it and address it in a later PR.